http-server-tck/src/main/java/io/micronaut/http/server/tck/tests/cors/CrossOriginTest.java

```diff
@@ -0,0 +1,278 @@
+ * limitations under the License.
+ */
+package io.micronaut.http.server.tck.tests.cors;
+
+import io.micronaut.context.annotation.Replaces;
+import io.micronaut.context.annotation.Requires;
+import io.micronaut.core.annotation.Nullable;
+import io.micronaut.http.HttpHeaders;
+import io.micronaut.http.HttpMethod;
+import io.micronaut.http.HttpRequest;
+import io.micronaut.http.HttpResponse;
+import io.micronaut.http.HttpStatus;
+import io.micronaut.http.MediaType;
+import io.micronaut.http.MutableHttpRequest;
+import io.micronaut.http.annotation.Controller;
+import io.micronaut.http.annotation.Delete;
+import io.micronaut.http.annotation.Get;
+import io.micronaut.http.annotation.PathVariable;
+import io.micronaut.http.annotation.Post;
+import io.micronaut.http.annotation.Produces;
+import io.micronaut.http.server.cors.CrossOrigin;
+import io.micronaut.http.server.tck.AssertionUtils;
+import io.micronaut.http.server.tck.HttpResponseAssertion;
+import io.micronaut.http.server.util.HttpHostResolver;
+import io.micronaut.http.uri.UriBuilder;
+import jakarta.inject.Singleton;
+import org.junit.jupiter.api.Test;
+
+import java.io.IOException;
+
+import static io.micronaut.http.server.tck.TestScenario.asserts;
+import static org.junit.jupiter.api.Assertions.assertAll;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+@SuppressWarnings({
+    "java:S2259", // The tests will show if it's null
+    "java:S5960", // We're allowed assertions, as these are used in tests only
+    "checkstyle:MissingJavadocType",
+})
+public class CrossOriginTest {
+
+    private static final String SPECNAME = "CrossOriginTest";
+
+    @Test
+    void crossOriginAnnotationWithMatchingOrigin() throws IOException {
+        asserts(SPECNAME,
+            preflight(UriBuilder.of("/foo").path("bar"), "https://foo.com", HttpMethod.GET),
+            (server, request) -> AssertionUtils.assertDoesNotThrow(server, request, HttpResponseAssertion.builder()
+                .status(HttpStatus.OK)
+                .assertResponse(response -> {
+                    assertCorsHeaders(response, "https://foo.com", HttpMethod.GET);
+                    assertFalse(response.getHeaders().names().contains(HttpHeaders.ACCESS_CONTROL_ALLOW_HEADERS));
+                })
+                .build()));
+    }
+
+    @Test
+    void crossOriginAnnotationWithNoMatchingOrigin() throws IOException {
+        asserts(SPECNAME,
+            preflight(UriBuilder.of("/foo").path("bar"), "https://bar.com", HttpMethod.GET),
+            (server, request) -> AssertionUtils.assertThrows(server, request, HttpResponseAssertion.builder()
+                .status(HttpStatus.METHOD_NOT_ALLOWED)
+                .assertResponse(this::assertCorsHeadersNotPresent)
+                .build()));
+    }
+
+    @Test
+    void corsSimpleRequestMethods() {
```

Comment:
The way the docs read. It seems that still you have to enable  CORS via configuration  to use `@CrossOrigin`.